extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/TransferProcessApiExtension.java

```diff
@@ -47,16 +50,18 @@ public class TransferProcessApiExtension implements ServiceExtension {
     @Inject
     private TransactionContext transactionContext;
 
-
     @Override
     public String name() {
         return "Data Management API: Transfer Process";
     }
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        webService.registerResource(configuration.getContextAlias(), new TransferProcessApiController(context.getMonitor(),
-                new TransferProcessServiceImpl(transferProcessStore, manager, transactionContext), transformerRegistry));
+        var service = new TransferProcessServiceImpl(transferProcessStore, manager, transactionContext);
+        context.registerService(TransferProcessService.class, service);
```

Comment:
any reason not to do this via the `@Provider` method?

Reply:
@paullatzelsperger I didn't thought about that, however I have some issues with that:

if I provide it with:
```java
@Provider
public TransferProcessService transferProcessService() {
    return new TransferProcessServiceImpl(transferProcessStore, manager, transactionContext);
}
```
I get a 
```
org.eclipse.dataspaceconnector.spi.system.injection.EdcInjectionException: No default provider for required service interface org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.service.TransferProcessService
```

if I provide as default with:
```java
@Provider(isDefault = true)
public TransferProcessService transferProcessService() {
    return new TransferProcessServiceImpl(transferProcessStore, manager, transactionContext);
}
```
I get a 
```
java.lang.NullPointerException
	at org.eclipse.dataspaceconnector.api.datamanagement.transferprocess.service.TransferProcessServiceImpl.initiateTransfer(TransferProcessServiceImpl.java:79)
```
seems like the `transactionContext` is not initialized for some reason (maybe because it is a default instance too?)